packages/router/src/router.ts

```diff
@@ -287,9 +297,12 @@ export class Router {
 
     this.replacePaths(instruction);
 
-    if (!keepHistoryEntry) {
+    // Remove history entry if no history viewports updated
+    if (!instruction.isFirst && !updatedViewports.reduce((accumulated: boolean, current: Viewport) => !current.options.noHistory || accumulated, instruction.isFirst)) {
       this.historyBrowser.pop().catch(error => { throw error; });
     }
+
+    updatedViewports.forEach((value) => value.finalizeContentChange());
```

Comment:
```suggestion
    updatedViewports.forEach(viewport => {
      viewport .finalizeContentChange()
    });
```

Makes it easier to debug.